File: VHLNavigation/ViewController.m
//
//  ViewController.m
//  VHLNavigation
//
//  Created by vincent on 2017/8/28.
//  Copyright © 2017年 Darnel Studio. All rights reserved.
//

#import "ViewController.h"
#import "FakeNavViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"导航栏";
    self.view.backgroundColor = [UIColor whiteColor];
    
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 150, 100)];
    [button setTitle:@"导航栏样式" forState:UIControlStateNormal];
    [button setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
    [button setBackgroundColor:[UIColor blackColor]];
    [self.view addSubview:button];
    button.center = self.view.center;
    [button addTarget:self action:@selector(gonext:) forControlEvents:UIControlEventTouchUpInside];
    
}
/** prefersLargeTitles 大标题显示 */
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    if (@available(iOS 11.0, *)) {
        self.navigationController.navigationBar.prefersLargeTitles = YES;
        self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeAutomatic;
    }
}
- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    if (@available(iOS 11.0, *)) {
        self.navigationController.navigationBar.prefersLargeTitles = NO;
         self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeAutomatic;
    }
}
- (void)gonext:(UIButton *)sender {
    FakeNavViewController *vc1 = [[FakeNavViewController alloc] init];
    [self.navigationController pushViewController:vc1 animated:YES];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Demo_swift/VHLNavigation_Swift/ViewController.swift
//
//  ViewController.swift
//  VHLNavigation_Swift
//
//  Created by Vincent on 2019/9/28.
//  Copyright © 2019 Darnel Studio. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let button = UIButton(frame: CGRect(x: 100, y: 60 + 64, width: 150, height: 30))
        button.setTitle("微信样式", for: .normal)
        button.setTitleColor(.white, for: .normal)
        button.backgroundColor = .black
        self.view.addSubview(button)
        
        button.addTarget(self, action: #selector(buttonClick), for: .touchUpInside)
    }

    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
    }
    
    @objc func buttonClick() {
        let fakeVC = FakeViewController()
        self.navigationController?.pushViewController(fakeVC, animated: true)
    }
}

